### PR TITLE
BUG: `interpolate` with fillna methods fail to fill across multiblocks

### LINKED ISSUE #53898: BUG: interpolate with ffill/bfill method, axis=1, multi-block
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import numpy as np
import pandas as pd

df = pd.DataFrame(np.random.randn(3, 3), columns=["A", "B", "C"])
df["D"] = np.nan
df["E"] = 1.0

>>> df.interpolate(method="ffill", axis=1)
<stdin>:1: FutureWarning: DataFrame.interpolate with method=ffill is deprecated and will raise in a future version. Use obj.ffill() or obj.bfill() instead.
          A         B         C   D    E
0 -0.656239  0.898067  0.842284 NaN  1.0
1 -0.914892 -0.018121 -0.382542 NaN  1.0
2  1.818251 -0.148962  0.550328 NaN  1.0
```

### Issue Description

This operates block-by-block and fails to fill across blocks.

I'd expect this to behave like `df.T.interpolate(method="ffill").T`  That is what we do in NDFrame._pad_or_backfill.  Note that it doesn't play nicely with the "downcast" keyword.

This value for "method" is deprecated, but still might be worth doing something about in the interim.

### Expected Behavior

NA

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

### PR DESCRIPTION
**Pull request by Charlie-XIAO**
- [ ] Closes #53898
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests)
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit)
- [x] Added an entry in the latest `doc/source/whatsnew/v2.1.0.rst` file

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -7968,9 +7968,17 @@ def interpolate(
         axis = self._get_axis_number(axis)
 
         fillna_methods = ["ffill", "bfill", "pad", "backfill"]
-        should_transpose = axis == 1 and method not in fillna_methods
 
-        obj = self.T if should_transpose else self
+        if method not in fillna_methods:
```
Comment (jbrockmendel) [inline pandas/core/generic.py:7972]: can you do this down inside the pad_or_backfill block on L8023?

Another option would be to dispatch to self._pad_or_backfill, but that would require adding a limit_area keyword there (which might be worthwhile long-term anyway)

Comment (Charlie-XIAO) [reply pandas/core/generic.py]: > can you do this down inside the pad_or_backfill block on L8023?

Sure, pushed a commit.

> Another option would be to dispatch to self._pad_or_backfill, but that would require adding a limit_area keyword there (which might be worthwhile long-term anyway)

Will try to investigate, and maybe open another PR for this.

File: pandas/core/generic.py [hunk 2]
```diff
@@ -7988,9 +7985,9 @@ def interpolate(
                 FutureWarning,
                 stacklevel=find_stack_level(),
             )
-        elif np.any(obj.dtypes == object):
+        elif np.any(self.dtypes == object):
```
Comment (jbrockmendel) [inline pandas/core/generic.py:7988]: i think this actually needs to be the maybe-transposed object for the interpolate check to be correct.

Comment (Charlie-XIAO) [reply pandas/core/generic.py]: > i think this actually needs to be the maybe-transposed object for the interpolate check to be correct.

I reorganized the code a bit so as to fix this check. I moved one of the deprecations warnings down, but I think it doesn't matter.

File: pandas/core/generic.py [hunk 3]
```diff
@@ -8050,6 +8041,17 @@ def interpolate(
                 downcast=downcast,
             )
         else:
+            obj, should_transpose = (self.T, True) if axis == 1 else (self, False)
+            if np.any(obj.dtypes == object):
```
Comment (jbrockmendel) [inline pandas/core/generic.py:8045]: this is nitpicky, but doing this after the self.empty check means we wont warn in some cases

File: pandas/core/generic.py [hunk 4]
```diff
@@ -8023,23 +8028,22 @@ def interpolate(
...
-            raise ValueError(
-                "'fill_value' is not a valid keyword for "
-                f"{type(self).__name__}.interpolate"
-            )
-
         if method.lower() in fillna_methods:
             # TODO(3.0): remove this case
             # TODO: warn/raise on limit_direction or kwargs which are ignored?
-            #  as of 2023-06-26 no tests get here with either
+            # as of 2023-06-26 no tests get here with either
```
Comment (jbrockmendel) [inline pandas/core/generic.py:8041]: the extra space after the "#" is intentional

**Review by jbrockmendel — APPROVED** [review]

File: pandas/core/generic.py (hunk 3 above)
Comment (Charlie-XIAO) [reply pandas/core/generic.py]: Fixed

File: pandas/core/generic.py (hunk 4 above)
Comment (Charlie-XIAO) [reply pandas/core/generic.py:8041]: Okay I'll revert

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @Charlie-XIAO

**Pull request merged**